omgeo/preprocessors.py

```diff
@@ -217,8 +223,14 @@ def __init__(self, regex, attrs, ignorecase=True):
 
     def process(self, pq):
         attrs = [getattr(pq, attr) for attr in self.attrs if hasattr(pq, attr)]
-        if any([self.regex.match(attr) is not None for attr in attrs]):
-            return False  # if a match is found
+        for attr in attrs:
+            # Handle string type mismatch by decoding if the first attempt raises TypeError
+            try:
+                match = self.regex.match(attr)
+            except TypeError:
+                match = self.regex.match(attr.decode())
```

Comment:
This should be `attr.decode('utf-8')`; in Python 3, `'utf-8'` is the default encoding used with `bytes.decode()`, but in Python 2.7 the default encoding is `'ascii'`. I suspect that this will fail under Python 2.7 when passed a non-Latin bytestring.

Reply:
Oops, so indeed.  Though I'm not sure that failure case can happen in practice.  Python 2 seems to be more accepting of mismatch between the regex pattern string type and the target string type, so I've only managed to trigger that `TypeError` in Python 3.  I managed to get `attr` values that would crash in Python 2 if you called `.decode()` on them, but since they didn't raise `TypeError` regardless of how the regex was formatted, they didn't get to that point.